subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -368,15 +368,6 @@ property has been deprecated.
 Use the link:{groovyDslPath}/org.gradle.api.tasks.scala.ScalaCompile.html#org.gradle.api.tasks.scala.ScalaCompile:destinationDirectory[ScalaCompile.destinationDirectory]
 property instead.
 
-[[deprecated_flat_project_structure]]
-==== Deprecated flat project structure
```

Comment:
I wonder about keeping something in here. Because if folks update to 7.0 -> 7.3.1, they will get this warning. We should indicate that 7.4 removes it.

Reply:
I've added documentation to the upgrade guide.